### ISSUE TITLE
Unsure how to get Unifi Controller behind Zoraxy

### ISSUE THREAD

**Issue by gokou340** [tobychui/zoraxy#115]
I'm in the process of migrating from NPM to Zoraxy and the last thing I need to figure out is how to get my Unifi Controller to work behind Zoraxy.

When Zoraxy is being used and I put in my username and password into Unifi, I get the following: "There was an error making that request. Please try again later."

In NPM and Nginx, I had to add the following previously:
```
        }
        location / {
                proxy_pass https://IP:8443/;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header Host $http_host;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "Upgrade";
                proxy_buffering off;
                proxy_ssl_verify off;
                        }
```

I'm not familiar enough with Zoraxy to understand how to add these to the connection. I'd like to fully migrate to Zoraxy, and this is my last step.

Any input would be appreciated!

**Comment by tobychui** [comment]
@gokou340 Would you mind showing me how you setup the proxy for your Controller in Zoraxy? 
You will find the config under conf/proxy/*.config in JSON format that corresponding to the HTTP proxy rule.

**Comment by gokou340** [comment]
@tobychui  Sure thing! Here you go:
`{
 "ProxyType": 1,
 "RootOrMatchingDomain": "unifi.ctulab.com",
 "Domain": "10.192.200.51:8443",
 "RequireTLS": true,
 "BypassGlobalTLS": false,
 "SkipCertValidations": true,
 "VirtualDirectories": [],
 "UserDefinedHeaders": [],
 "RequireBasicAuth": false,
 "BasicAuthCredentials": [],
 "BasicAuthExceptionRules": [],
 "DefaultSiteOption": 0,
 "DefaultSiteValue": "",
 "Disabled": false
`

**Comment by tobychui** [comment]
@gokou340 From your config, I see no issues with the proxy setup. 
Zoraxy automatically adds all the required headers to the proxy rule by sniffing HTTP headers and MIME types. So the following headers should be automatically configured and seems SkipCertValidations is set correctly as well.
```
X-Forwarded-For
X-Forwarded-Proto
X-Real-IP
Upgrade
Connection
```
Currently zoraxy have no support for ```proxy_buffering off```. But base on your use case, I am adding auto detection for non-streaming responses in the upcoming v3.0.1 builds.

**Comment by gokou340** [comment]
That sounds good.

Running a trace, I can see that I am getting a 403 Forbidden message when I submit my credentials in the controller:
```
Request URL:
https://URL/api/login
Request Method:
POST
Status Code:
403 Forbidden
Remote Address:
ZORAXYIP:443
Referrer Policy:
strict-origin-when-cross-origin
```

Let me know if this is something different than what I am thinking.

**Comment by gokou340** [comment]
Also, to add, here are some posts about the same error when behind Apache/Nginx:
https://community.ui.com/questions/Controller-v5-11-behind-reverse-proxy-gives-403-on-login/f98c3c5c-74c2-4cd5-80c1-24af498940aa

https://community.ui.com/questions/Controller-behind-Apache-Reverse-Proxy-greater-login-does-not-work/1925d1f7-1b95-4ca6-94dd-2161dcbf77e7

It might help with debugging. If you need me to pull any logs or do anything on my end, let me know!

**Comment by gokou340** [comment]
@tobychui 
Are you asking me to just create an A Record in my local DNS that points to my Unifi Controller, or are you saying to create something within Zoraxy?

**Comment by tobychui** [comment]
@gokou340 The DNS A record, and use the domain name (instead of ip address) to connect to your Unifi controller to see if the same error will be shown.

**Comment by tobychui** [comment]
@gokou340 You can send me an email at imuslab@gmail.com and we can chat directly, but I am in GMT+8 so probably we don't got much time that we can chat directly.

**Comment by tobychui** [comment]
@AmIBeingObtuse We are still working on it. We are really lacking on contributors to keep up with the issues with different open source projects being proxied so this issue might take a while to be handled. 

Btw, are you asking because you have tried out the v3.0.5 and have the same issue mentioned here? More inputs and sharing of debug attempts helps us fix issues faster :)

**Comment by AmIBeingObtuse** [comment]
Hi

Wish I had checked it before commenting. It works for me! I have my unifi
controller on unifi.myinternaldomain.com and works no issues (could never
get this working on NPM).

Just setup at a standard proxy with my wildcard ssl cert and its working.

also this is my first time with Zoraxy and I am blown away. I will be doing
a youtube video on this. Thanks! ***@***.*** in the next
week or so. One of my favourite features is that you provide a windows
binary.

**Comment by tobychui** [comment]
Ok, then I guess I can close this now as the header bug fix in v3.0.5 also fixed this issue.

**Comment by joakimkingstrom** [comment]
I had issues with getting Unifi to work with Zoraxy as well, but in my particular case it was solved by changing the destination from using a dns name to using the IP address instead.
